Twilio/Tests/Unit/Http/GuzzleClientTest.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+
+namespace Twilio\Tests\Unit\Http;
+
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Handler\MockHandler;
+use GuzzleHttp\HandlerStack;
+use GuzzleHttp\Psr7\Response;
+use Twilio\Http\GuzzleClient;
+use Twilio\Tests\Unit\UnitTest;
+
+final class GuzzleClientTest extends UnitTest
+{
+    /**
+     * @var GuzzleClient
+     */
+    private $client;
+
+    /**
+     * @var MockHandler
+     */
+    private $mockHandler;
+
+    public function setUp()
+    {
+        parent::setUp();
+        $this->mockHandler = new MockHandler();
+        $this->client = new GuzzleClient(new Client([
+            'handler' => HandlerStack::create($this->mockHandler),
+        ]));
+    }
+
+    public function testPostMethod()
```

Comment:
A _4XX_ or _5XX_ test would help validate/document the `BadResponseException` logic.

Reply:
done :)